### PR TITLE
stop testing containers for pod tests

### PR DESCRIPTION
**Pull request by ydayagi**
https://issues.redhat.com/browse/CNFCERT-40
Signed-off-by: Yaron Dayagi <ydayagi@redhat.com>

### CODE REVIEW

File: test-network-function/generic/suite.go [hunk 1]
```diff
@@ -154,9 +154,10 @@ type container struct {
 func createContainers(containerDefinitions []configsections.Container) map[configsections.ContainerIdentifier]*container {
 	createdContainers := make(map[configsections.ContainerIdentifier]*container)
 	for _, c := range containerDefinitions {
+		podKey := configsections.GetPodIDFromContainerID(c.ContainerIdentifier)
 		oc := getOcSession(c.PodName, c.ContainerName, c.Namespace, defaultTimeout, interactive.Verbose(true))
 		var defaultIPAddress = "UNKNOWN"
-		if _, ok := containersToExcludeFromConnectivityTests[c.ContainerIdentifier]; !ok {
+		if _, ok := podsToExcludeFromConnectivityTests[podKey]; !ok {
```
[the remark this answers is not in the document]
Comment (jun-rh) [reply test-network-function/generic/suite.go:160]: You mean all the containers in a pod share the same network interface?

[the remark this answers is not in the document]
Comment (HamadiSE) [reply test-network-function/generic/suite.go:160]: my bad, right/ containers are within a pod. and its the POD who has the IP. my comment is not valid.

**Comment by jun-rh** [comment]
@ryandgoulding @crwr45 I am surprised to see this many pod level tests in generic. Does it makes more sense to move them to the "container" spec where other pod level tests are and rename the spec to something else?

**Review by HamadiSE — APPROVED** [review]

**Comment by ryandgoulding** [comment]
> @ryandgoulding @crwr45 I am surprised to see this many pod level tests in generic. Does it makes more sense to move them to the "container" spec where other pod level tests are and rename the spec to something else?

Dealer's choice.  Historically, the `container` spec was abstracted by a contributor to isolate it from existing test cases during development.  One could argue that some tests belong there, or to get rid of `container` altogether and combine it with `generic`.

**Review by crwr45 — CHANGES_REQUESTED** [review]
This looks like a step in the right direction. I'm lacking some of the context to know what followup changes are coming but assuming a more complete refactoring and restructuring around containers/pods, the code is good.

This does need an update to the documentation to ensure the new behaviour is clear.

**Comment by jun-rh** [comment]
Thanks @ryandgoulding @crwr45
We plan to step back and look at test categorization overall. Once we have a good strategy on that, we will design the config part accordingly (pod list vs container list, common or per suite etc).

**Comment by pkliczewski** [comment]
@ydayagi do we still want to finalize this PR?

**Pull request closed without merging**